.travis.yml

```diff
@@ -7,24 +7,25 @@ branches:
 jobs:
   fast_finish: true
   allow_failures:
-    - env: WEBPACK_VERSION=canary
+    - env: WEBPACK_VERSION=canary JOB_PART=test
   include:
     - &test-latest
-      stage: Webpack latest
+      stage: Webpack 3.8.1
       node_js: 6
-      env: WEBPACK_VERSION=latest JOB_PART=test
+      env: WEBPACK_VERSION=3.8.1 JOB_PART=test
       script: npm run travis:$JOB_PART
     - <<: *test-latest
       node_js: 4.8
-      env: WEBPACK_VERSION=latest JOB_PART=test
+      env: WEBPACK_VERSION=3.8.1 JOB_PART=test
       script: npm run travis:$JOB_PART
     - <<: *test-latest
       node_js: 8
-      env: WEBPACK_VERSION=latest JOB_PART=lint
+      env: WEBPACK_VERSION=3.8.1 JOB_PART=lint
+      before_script: npm i -g npm@^6.0.0 && npm --version
       script: npm run travis:$JOB_PART
     - <<: *test-latest
       node_js: 8
-      env: WEBPACK_VERSION=latest JOB_PART=coverage
+      env: WEBPACK_VERSION=3.8.1 JOB_PART=coverage
```

Comment:
Why `3`? We should drop support webpack@3 and prefer to use webpack@4

Reply:
webpack v4 drops support for Node 4 and use es6, it will fail as long as we want to support Node v4.

Is it OK to drop support for Node 4? It would make everything much easier and resolve almost all your comments as I can upgrade Webpack and babel and npm to latest.

Reply:
> Is it OK to drop support for Node 4?

yes feel free to drop node@4

Reply:
Could we get this PR through as it is definitely an improvement over the current code, and I can work on a separate PR to upgrade to Webpack 4 with the latest dependencies and re-using latest versions in the CI pipelines?

Reply:
Updated to latest dependencies and Webpack v4. Had to drop support for node 4 & 6 in the process

Reply:
Reverted the upgrade and wrote a separate comment about it